styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -633,21 +633,31 @@ private static DockerRunner createDockerRunner(
         serviceAccountKeyManager, debug, styxEnvironment, secretWhitelist));
   }
 
-  static FlyteRunner createFlyteRunner(String runnerId, Config config, StateManager stateManager){
+  static FlyteRunner createFlyteRunner(String runnerId, Config config, StateManager stateManager) {
     if (!config.getBoolean(FLYTE_ENABLED)) {
       return FlyteRunner.noop();
     }
 
+    var flyteAdminClient = getFlyteAdminClient(config, runnerId);
+    return new FlyteAdminClientRunner(runnerId, flyteAdminClient, stateManager);
+  }
+
+  private static FlyteAdminClient getFlyteAdminClient(Config rootConfig, String runnerId) {
+    var flyteAdminRootConfig = rootConfig.getConfig(FLYTEADMIN_PATH);
+    if (!flyteAdminRootConfig.hasPath(runnerId)) {
+      throw new IllegalArgumentException("There is no configuration for flyte admin for runner id: " + runnerId);
+    }
+
+    var config = flyteAdminRootConfig.getConfig(runnerId);
```

Comment:
Isn't this going to be the runnerId in datastore? i.e., staging-regional for staging? or am I missing something @narape @ckiosidis

Reply:
Aa now I see it 👍